packages/core/src/modules/proofs/protocol/v1/handlers/V1RequestPresentationHandler.ts

```diff
@@ -0,0 +1,127 @@
+import type { AgentConfig } from '../../../../../agent/AgentConfig'
+import type { Handler, HandlerInboundMessage } from '../../../../../agent/Handler'
+import type { DidCommMessageRepository } from '../../../../../storage/didcomm/DidCommMessageRepository'
+import type { MediationRecipientService } from '../../../../routing'
+import type { ProofResponseCoordinator } from '../../../ProofResponseCoordinator'
+import type { ProofService } from '../../../ProofService'
+import type { ProofRecord } from '../../../repository/ProofRecord'
+
+import { createOutboundMessage, createOutboundServiceMessage } from '../../../../../agent/helpers'
+import { ServiceDecorator } from '../../../../../decorators/service/ServiceDecorator'
+import { AriesFrameworkError } from '../../../../../error/AriesFrameworkError'
+import { DidCommMessageRole } from '../../../../../storage'
+import { ProofProtocolVersion } from '../../../models/ProofProtocolVersion'
+import { V1ProposePresentationMessage, V1RequestPresentationMessage } from '../messages'
+
+export class V1RequestPresentationHandler implements Handler {
+  private proofService: ProofService
+  private agentConfig: AgentConfig
+  private proofResponseCoordinator: ProofResponseCoordinator
+  private mediationRecipientService: MediationRecipientService
+  private didCommMessageRepository: DidCommMessageRepository
+  public supportedMessages = [V1RequestPresentationMessage]
+
+  public constructor(
+    proofService: ProofService,
+    agentConfig: AgentConfig,
+    proofResponseCoordinator: ProofResponseCoordinator,
+    mediationRecipientService: MediationRecipientService,
+    didCommMessageRepository: DidCommMessageRepository
+  ) {
+    this.proofService = proofService
+    this.agentConfig = agentConfig
+    this.proofResponseCoordinator = proofResponseCoordinator
+    this.mediationRecipientService = mediationRecipientService
+    this.didCommMessageRepository = didCommMessageRepository
+  }
+
+  public async handle(messageContext: HandlerInboundMessage<V1RequestPresentationHandler>) {
+    const proofRecord = await this.proofService.processRequest(messageContext)
+    if (this.proofResponseCoordinator.shouldAutoRespondToRequest(proofRecord)) {
+      return await this.createPresentation(proofRecord, messageContext)
+    }
+  }
+
+  private async createPresentation(
+    record: ProofRecord,
```

Comment:
for consistency

```suggestion
    proofRecord: ProofRecord,
```

Reply:
As the `proofRecord` is being used to store the proof record after the service sends the created presentation so not able to change it.